app/gui2/src/stores/graph/graphDatabase.ts

```diff
@@ -524,16 +559,18 @@ export class GraphDb {
   }
 }
 
-interface BlockLine {
+interface NodeInfo {
   outerAst: Ast.Ast
-  isLastInBlock: boolean
+  isOutput: boolean
+  isInput: boolean
+  argIndex: number | undefined
```

Comment:
It's not clear what is the difference between NodeInfo and Node. Maybe this should be "NodeSource"? (A line or parameter).

Reply:
True, thank you for suggestion, I was struggling with finding a better naming.